packages/scrapers-v2/src/classify/classify.ts

```diff
@@ -0,0 +1,47 @@
+import { ensureLengthAtLeast, loadHTML, parseText } from "../utils";
+import { CatalogEntryType, TypedCatalogEntry } from "./types";
+
+export const classifyCatalogEntries = async (
+  flattenedList: string[],
+): Promise<TypedCatalogEntry[]> => {
+  return await Promise.all(flattenedList.map(addTypeToUrl));
+};
+
+const addTypeToUrl = async (url: string): Promise<TypedCatalogEntry> => {
+  const $ = await loadHTML(url);
+  const type = getUrlType($);
+  return { url, type };
+};
+
+const getUrlType = ($: CheerioStatic): CatalogEntryType => {
+  const tabsContainer = $("#contentarea #tabs");
+
+  if (!tabsContainer || tabsContainer.length === 0) {
+    return CatalogEntryType.Unknown;
+  }
+
+  if (tabsContainer.length > 1) {
+    throw new Error(
+      `Expected 1 tab container, but found ${tabsContainer.length}.`
+    );
+  }
```

Comment:
just a thought, it would be nice if we followed either the pattern of throwing errors first and then handling our cases or handling our cases first and then throwing errors consistently. 

For instance, if we were doing errors before and then normal cases then we could have the check of `if (tabs > 3) throw` here instead of at the end. And then within the `tabs.len === 3` case we can do `if middleTabText !== "Program requirements" throw` first and then return instead of the other way around. 

Again, totally ok with handling normal cases first and then throwing errors, would just like to see 1 consistent pattern so that it's easy to follow through the different cases

Reply:
yep, refactored